nav2_costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -167,17 +167,7 @@ void VoxelLayer::matchSize()
 
 void VoxelLayer::reset()
 {
-  voxel_pub_->on_deactivate();
-  deactivate();
-  resetMaps();
-  activate();
-  undeclareAllParameters();
-  voxel_pub_->on_activate();
-}
-
-void VoxelLayer::resetMaps()
```

Comment:
Where did the `resetMaps` method in Voxel layer go? It looks like you replaced it work only `reset()`, why?

Reply:
The relevant inheritance hierarchy looks like this:

![image](https://user-images.githubusercontent.com/39749483/70739605-0f128700-1ccc-11ea-8500-a7d8d49414bf.png)

Voxel layer was overriding the Costmap2D method only to call it directly without adding any functionality. It served no purpose.

What's more, if obstacle layer at sometime decided to override resetMaps, VoxelLayer would have bypassed it.

Reply:
Let me add, resetMaps is the method to reset the Costmap2D base class and reset is the method to reset the Layer base class.

They are two separate methods in the interface. We now override reset in VoxelLayer and no longer override resetMaps.

It's not a matter of changing the method name.

![image](https://user-images.githubusercontent.com/39749483/70739997-ff477280-1ccc-11ea-909a-46ce53116610.png)

Reply:
This maybe is a slightly better diagram
![image](https://user-images.githubusercontent.com/39749483/70741349-ea201300-1ccf-11ea-8883-c2a30971ec13.png)

Reply:
That's untrue, see here https://github.com/ros-planning/navigation2/blob/master/nav2_costmap_2d/plugins/voxel_layer.cpp#L121. The reset method is what is actually clearing the voxel grid data structure. 

`resetMaps` is then called by the costmap 2d layer in the rolling costmap mode which will clear out the rolling window of voxels as well for the current set.